### PR TITLE
Validated input dialog

### PR DESCRIPTION
**Pull request by lewisrenfrew**
- new component... similar to the TypeaheadDialog except instead of showing search results this components just gives feedback as to whether the thing under search exists or not. For example as an 'Enter Works Order' Dialog the component will go red and not allow you to accept if you enter an invalid worksOrderNumber or Green and allow you to acceptif you enter a valid one. I think this'll come in quite handy but can demo it at show n tell if there are doubts.

- also fixed various story issues introduced by notistack and themes - defining a common provider function that wraps stories in all the gubbings they need to not error

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
         "prettier": "^1.18.2",
         "prop-types": "^15.7.2",
         "query-string": "^6.5.0",
+        "react-redux": "^7.1.1",
```
Comment (atcampbell) [inline package.json]: Is it ok to have this here and in peer dependencies? (I don't really know how they work...)

File: src/components/ValidatedInputDialog.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        setDialogOpen(false);
+    };
+
+    const handleSearchTermChange = (...args) => {
+        setSearchTerm(args[1]);
+    };
+
+    const getTheme = () => {
+        if (!searchTerm || loading) return linnTheme;
+        return searchItems.length === 1 ? valid : invalid;
```
Comment (atcampbell) [inline src/components/ValidatedInputDialog.js]: Maybe these variables would be clearer as validTheme etc? Also maybe not

File: package.json (hunk 1 above)
Comment (lewisrenfrew) [reply package.json]: yeah... annoyingly storybook was complaining about not having an instance of react-redux to play with. The only workaround I could find for now was installing it in here as well. 

Good spot though - this does cause problems for consuming projects which needs to be addressed in their webpack config aliases... technically making this a breaking change. I'll see if I can find a workaround before deploying.

**Comment by lewisrenfrew** [comment]
> I think this is a really good idea. Does it need to be done in a modal? It might be nice to have an option to do this as an input field inline - that could happen while in the flow of updating the document. Maybe that way users don't need to think of it as a search, but just as validation going on in the background?

Yeah maybe you're right.. might be a nicer user experience not to bother with a modal. Do you think it's worth having both versions available as in the case of the Typeahead? Or should I just change this to ditch the modal.

**Comment by atcampbell** [comment]
> > I think this is a really good idea. Does it need to be done in a modal? It might be nice to have an option to do this as an input field inline - that could happen while in the flow of updating the document. Maybe that way users don't need to think of it as a search, but just as validation going on in the background?
> 
> Yeah maybe you're right.. might be a nicer user experience not to bother with a modal. Do you think it's worth having both versions available as in the case of the Typeahead? Or should I just change this to ditch the modal.

I'm not sure really, if there's a benefit and use of the modal version then we should have both. 

Maybe it could be an input field with its own loading indicator to show that something is happening in the background, so that it still gets across that its a search (something like this underneath perhaps https://material-ui.com/components/progress/#linear-query)?

**Comment by atcampbell** [comment]
Oh and you should update the changelog 😉

**Pull request merged**